src/index.js

```diff
@@ -81,20 +86,40 @@ class NetworkStateNotifier extends React.Component {
     });
   };
 
-  handleChecker = () => {
-    const { isOnline } = this.state;
+  toggleOnlineOffline = (status) => {
     const { notificationTimeout } = this.props;
+    const { isOnline } = this.state;
 
-    if (isOnline !== navigator.onLine) {
-      this.handleMessageAddition();
-      this.setState(
-        {
-          isOnline: navigator.onLine,
-        },
-        () => {
-          setTimeout(this.handleMessageRemove, notificationTimeout);
-        },
-      );
+    if (isOnline === status) return;
+
+    this.handleMessageAddition();
+    this.setState(
+      {
+        isOnline: status,
+      },
+      () => {
+        setTimeout(this.handleMessageRemove, notificationTimeout);
+      },
+    );
+  };
+
+  handleChecker = () => {
+    const { polling, isOnline } = this.state;
+
+    if (polling) {
+      fetch(pollingURL, {}, 2000)
+        .then((res) => { // eslint-disable-line
+          if (!isOnline) {
```

Comment:
Don't you need to check the status of the response?

Reply:
I believe as long as there is a response wouldn't that mean that host is connected to the internet?

Reply:
My bad. Missed that point.